agent/agent.go

```diff
@@ -721,6 +730,11 @@ func (a *Agent) runPreAndMain(ctx context.Context, tc *taskContext) (status stri
 		}
 	}
 
+	defer func() {
```

Comment:
Do we need to defer a `os.RemoveAll` earlier (in case like `runPreTaskCommands` returns `TaskFailed`) so we'll be sure the directory gets removed?

Reply:
That makes sense. I moved it to the same defer statement where we upload the OTel traces from the reserved OTel directory. I also created [DEVPROD-5156](https://jira.mongodb.org/browse/DEVPROD-5156) to consolidate the task output handling by moving the otel file upload logic into the `agent/internal/taskoutput` package.